File: csharp/lesson 6/LevelOrder.cs
// https://leetcode.com/problems/binary-tree-level-order-traversal/

public class Solution
{
    public IList<IList<int>> LevelOrder(TreeNode root)
    {
        var result = new List<IList<int>>();
        if (root == null) return result;

        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);

        while (queue.Count > 0)
        {
            int size = queue.Count;
            var level = new List<int>();
            for (int i = 0; i < size; i++)
            {
                var node = queue.Dequeue();
                level.Add(node.val);
                if (node.left != null) queue.Enqueue(node.left);
                if (node.right != null) queue.Enqueue(node.right);
            }
            result.Add(level);
        }

        return result;
    }
}

File: javascript/lesson 6/levelOrder.js
// https://leetcode.com/problems/binary-tree-level-order-traversal/

function levelOrder(root) {
    const result = [];
    if (!root) return result;
 
    const queue = [root];
     while (queue.length > 0) {
        const size = queue.length;
        const level = [];
        for (let i = 0; i < size; i += 1) {
            const node = queue.shift();
            level.push(node.val);
            if (node.left) queue.push(node.left);
            if (node.right) queue.push(node.right);
        }
        result.push(level);
    }
    return result;
}
